src/Stratis.Features.FederatedPeg.IntegrationTests/StratisXSyncTests.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using NBitcoin;
+using Stratis.Bitcoin.Features.RPC;
+using Stratis.Bitcoin.IntegrationTests.Common.EnvironmentMockUpHelpers;
+using Stratis.Bitcoin.IntegrationTests.Common.ReadyData;
+using Stratis.Bitcoin.Networks;
+using Stratis.Bitcoin.Tests.Common;
+using Stratis.Features.FederatedPeg.IntegrationTests.Utils;
+using Stratis.Sidechains.Networks;
+using Xunit;
+
+namespace Stratis.Features.FederatedPeg.IntegrationTests
+{
+    public class StratisXSyncTests
+    {
+        private readonly CirrusRegTest sidechainNetwork;
+        private readonly Network mainNetwork;
+
+        private readonly (Script payToMultiSig, BitcoinAddress sidechainMultisigAddress, BitcoinAddress mainchainMultisigAddress) scriptAndAddresses;
+
+
+        public StratisXSyncTests()
+        {
+            this.sidechainNetwork = (CirrusRegTest)CirrusNetwork.NetworksSelector.Regtest();
+            this.mainNetwork = new StratisMain15KCheckpoint();
+            var pubKeysByMnemonic = this.sidechainNetwork.FederationMnemonics.ToDictionary(m => m, m => m.DeriveExtKey().PrivateKey.PubKey);
+            this.scriptAndAddresses = FederatedPegTestHelper.GenerateScriptAndAddresses(this.mainNetwork, this.sidechainNetwork, 2, pubKeysByMnemonic);
+        }
+
+        [Fact]
+        public void SyncFirst15KBlocks()
+        {
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                // TODO: Add the necessary executables for Linux & OSX
```

Comment:
I guess better to remove the TODO and add a comment that this is windows-only test. 
If you are not going to add those executables in this PR then it will never be done.

Reply:
I copied this straight from all of the other StratisX tests, can edit though